feat: add onLongPressUp callback to InkWell widget

houssemeddinefadhli18 opened the pull request:
This PR introduces a new optional callback, onLongPressUp, to the InkWell widget.
It allows developers to respond specifically to the moment when a long press gesture is released, which previously was not directly exposed.

**Before**
There was no way to distinguish between the long press being held and the moment it ended.

**After**
InkWell now accepts an onLongPressUp callback, which fires when the user lifts their finger after a long press gesture.
This change enhances gesture handling granularity for widgets that require different behaviors for long press hold and release.


**Related Issue**
#173390

**Tests**
- Added a new test to ink_well_test.dart verifying that onLongPressUp is correctly triggered after a long press is released.

**Checklist**
- I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- I read the [Tree Hygiene] wiki page.
- I followed the [Flutter Style Guide].
- I signed the [CLA].
- I updated/added relevant documentation (/// doc comments).
- I added new tests to check the change I am making.
- All existing and new tests are passing.

Commit:
feat: add onLongPressUp callback to InkWell widget

Commit:
fix: improve onLongPressUp documentation and correct typo in widget test

houssemeddinefadhli81 commented on packages/flutter/lib/src/material/ink_well.dart:
```diff
@@ -1230,6 +1238,14 @@ class _InkResponseState extends State<_InkResponseStateWidget>
+    if (widget.onLongPressUp != null) {
+      widget.onLongPressUp!();
+    }
+  }
```
Thanks for your reply. onLongPressUp does not provide any LongPressUpDetails; for example, onLongPressEnd has this signature.

Commit:
fix: improve onLongPressUp documentation and correct typo in widget test

victorsanni reviewed (commented):
Hi @houssemeddinefadhli81, thanks for the PR.

I don't see any linked issues, could you attach an existing one to this PR or file a new issue with a use case example for which adding this new API is the optimal solution?

Thanks!

Commit:
fix: improve onLongPressUp call safety with null-aware operator

houssemeddinefadhli81 commented:
> Hi @houssemeddinefadhli81, thanks for the PR.
> 
> I don't see any linked issues, could you attach an existing one to this PR or file a new issue with a use case example for which adding this new API is the optimal solution?
> 
> Thanks!

Issue : https://github.com/flutter/flutter/issues/173390

Commit:
fix: Update onLongPressUp documentation and upgrade Flutter SDK

houssemeddinefadhli81 commented:
@victorsanni @dkwingsmt Hello, I noticed a few CI checks have failed, but I’ve only made changes to two files: ink_well.dart and its corresponding test ink_well_test.dart. I’ve also rebased my branch on the latest master and re-ran all tests locally, including text_field_test.dart—and they all passed successfully. Please let me know if I should address anything specific or if a CI rerun would be appropriate. Thanks.

engine-flutter-autoroll committed:
Roll Fuchsia Linux SDK from 2Mo76-tiDj4ZCZbPv... to i4vsuEGyP8Xeb5tiy... (#173409)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/fuchsia-linux-sdk-flutter
Please CC codefu@google.com,zra@google.com on the revert to ensure that
a human
is aware of the problem.

To file a bug in Flutter:
https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

reidbaker committed:
[android][tool] Consolidate minimum versions for android projects.  (#171965)

I discovered we had a duplicate entry for what the min sdk api level
flutter supports along with duplicate code for telling the user.
This pr keeps the duplicate warnings but removed duplicate definitions. 
Updates the related tests
Updates workflow_test and android_workflow to compare against the
compile sdk version since the validation was ensuring that the lastest
of the installed sdks was at least what was needed for flutter to
compile. This make this pr more than a formatting/refactoring change and
an actual user facing change. Requesting re-review.
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description
above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [x] All existing and new tests are passing.

9AZX committed:
Provide monitor list, display size, refresh rate, and more for Windows (#164460)

This PR enhances the Windows implementation by adding support for
retrieving display properties, available displays, refresh rate, dpi,
size, and more.

Fixes: #160660, #125939

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description
above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [X] All existing and new tests are passing.

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

---------

Co-authored-by: Matthew Kosarek <matt.kosarek@canonical.com>

engine-flutter-autoroll committed:
Roll Packages from f0b272645902 to 6efb75994fc0 (4 revisions) (#173414)

https://github.com/flutter/packages/compare/f0b272645902...6efb75994fc0

2025-08-06 43054281+camsim99@users.noreply.github.com
[camera_android_camerax] Re-land "Force new Surface for each
SurfaceRequest" (flutter/packages#9760)
2025-08-06 filiph@users.noreply.github.com [google_maps_flutter] Add
ability to perform Google Maps SDK warmup (flutter/packages#9674)
2025-08-06 engine-flutter-autoroll@skia.org Roll Flutter from
59fc766c6fdf to 9de63a03428f (6 revisions) (flutter/packages#9764)
2025-08-06 gavrilko.alexander@gmail.com [webview_flutter_wkwebview]
Extended Web View API on iOS to add flexibility when working with local
HTML content (flutter/packages#8787)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-packages-flutter-autoroll
Please CC flutter-ecosystem@google.com on the revert to ensure that a
human
is aware of the problem.

To file a bug in Flutter:
https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

gaaclarke committed:
Adds deprecation for impeller opt out on android (#173375)

fixes https://github.com/flutter/flutter/issues/173122

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description
above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel
on [Discord].

**Note**: The Flutter team is currently trialing the use of [Gemini Code
Assist for
GitHub](https://developers.google.com/gemini-code-assist/docs/review-github-code).
Comments from the `gemini-code-assist` bot should not be taken as
authoritative feedback from the Flutter team. If you find its comments
useful you can update your code accordingly, but if you are unsure or
disagree with the feedback, please feel free to wait for a Flutter team
member's review for guidance on which automated comments should be
addressed.

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

micaelcid committed:
[web] add --static-assets-url argument to build web (#171638)

Closes https://github.com/flutter/flutter/issues/171637

As mentioned in the issue, I am proposing a new argument to the flutter
build web command: `--static-assets-url`.

This argument would accept a full URL string ending with `/` as its
value. During the build process, the Flutter tool would use this value
to replace a dedicated placeholder within the `web/index.html` file
(inspired by `--base-href` approach).

Example Implementation:

A developer would modify their `web/index.html` to use a new
placeholder, for instance, `$FLUTTER_STATIC_ASSETS_URL`:

```html
...
<body>
  <script>
    {{flutter_js}}
    {{flutter_build_config}}

    _flutter.loader.load({
      config: {
        entryPointBaseUrl: "$FLUTTER_STATIC_ASSETS_URL",
      },
      onEntrypointLoaded: async function (engineInitializer) {
        const appRunner = await engineInitializer.initializeEngine({
          assetBase: "$FLUTTER_STATIC_ASSETS_URL",
        });

        await appRunner.runApp();
      },
    });
  </script>
</body>
...
 ```

The build command would be run with the new flag: `flutter build web --static-assets-url="https://static.company.com/some-webapp/“` - and the resulting `build/web/index.html` would have the placeholder replaced: 

```html
...
<body>
  <script>
    {{flutter_js}}
    {{flutter_build_config}}

    _flutter.loader.load({
      config: {
        entryPointBaseUrl: "https://static.company.com/some-webapp/",
      },
      onEntrypointLoaded: async function (engineInitializer) {
        const appRunner = await engineInitializer.initializeEngine({
          assetBase: "https://static.company.com/some-webapp/",
        });

        await appRunner.runApp();
      },
    });
  </script>
</body>
...
```

victorsanni commented on packages/flutter/lib/src/material/ink_well.dart:
```diff
@@ -364,6 +365,20 @@ class InkResponse extends StatelessWidget {
+  /// This callback is triggered after the user has held a long press and then lifts their finger,
+  /// indicating the end of the long press interaction. It does not provide position details.
+  ///
+  /// To detect the position where the long press was released, consider using [onLongPressEnd] instead.
```
CI is failing because InkResponse doesn't have a `onLongPressEnd` member.

victorsanni commented on packages/flutter/test/material/ink_well_test.dart:
```diff
@@ -35,6 +36,9 @@ void main() {
               onLongPress: () {
                 log.add('long-press');
               },
+              onLongPressUp: () {
```
There are other places in this file where this log is verified. Can you add `long-press-up` to the log in those places?

victorsanni replied:
This test keeps failing, can you add `long-press-up` to the `expect`s below? i.e 

```dart
expect(log, equals(<String>['tap-down', 'tap-up', 'tap']));
```

houssemeddinefadhli81 replied:
Yes, that’s right, it’s done. Thanks!

victorsanni commented on packages/flutter/lib/src/material/ink_well.dart:
```diff
@@ -1230,6 +1249,14 @@ class _InkResponseState extends State<_InkResponseStateWidget>
+  void handleLongPressUp() {
+    _currentSplash?.confirm();
+    _currentSplash = null;
+    if (widget.onLongPressUp != null) {
```
Is this check still needed if the function is called conditionally anyways?

aam committed:
Manual roll to 3.10.0-75.1.beta (#173423)

Manual roll to 3.10.0-75.1.beta

```
6fe971b4c9b (HEAD, origin/lkgr-dev) Version 3.10.0-75.0.dev
c5eba845382 [analysis_server] Dot shorthands: Code completion for methods and constructors.
0cf15f211a4 [analysis_server] Fix change_method_signature tests to normalize code
62caf711fa1 Bump analyzer-related package version numbers
d4101733532 Ignore TODOs in analyzer_utilities package.
f2c73a7a3f0 [analysis_server] Fix DTD test EOLs on Windows
1cf804e1fe1 [vm, compiler] Fix Load[D]FromOffset for offsets with bits 21-23 set on ARM.
```

Commit:
fix: Add onLongPressUp in missing test cases in inkwell_test, remove redundant null check in InkWell, and update the onLongPress comment

Commit:
Merge branch 'master' into add-onLongPressUp-to-inkwell

victorsanni commented on packages/flutter/lib/src/material/ink_well.dart:
```diff
@@ -364,6 +365,19 @@ class InkResponse extends StatelessWidget {
 
+  /// Called when the user lifts their finger after a long press on the button.
+  ///
+  /// This callback is triggered at the end of a long press gesture—specifically,
```
```suggestion
  /// This callback is triggered at the end of a long press gesture, specifically
```

Commit:
Update packages/flutter/lib/src/material/ink_well.dart

fix: update onLongPressUp comment

Co-authored-by: Victor Sanni <victorsanniay@gmail.com>

Commit:
Update packages/flutter/lib/src/material/ink_well.dart

fix: update onLongPressUp comment

Co-authored-by: Victor Sanni <victorsanniay@gmail.com>

Commit:
Merge branch 'master' into add-onLongPressUp-to-inkwell

Commit:
Fix: Add onLongPressUp to the expected events in the InkWell test

Commit:
Merge branch 'add-onLongPressUp-to-inkwell' of https://github.com/houssemeddinefadhli81/flutter into add-onLongPressUp-to-inkwell

Commit:
Merge branch 'master' into add-onLongPressUp-to-inkwell

Commit:
fix: format code files

Commit:
Merge branch 'add-onLongPressUp-to-inkwell' of https://github.com/houssemeddinefadhli81/flutter into add-onLongPressUp-to-inkwell

Commit:
Merge branch 'master' into add-onLongPressUp-to-inkwell

victorsanni reviewed (approved):
LGTM pending CI

dkwingsmt commented on packages/flutter/lib/src/material/ink_well.dart:
```diff
@@ -1230,6 +1248,12 @@ class _InkResponseState extends State<_InkResponseStateWidget>
+  void handleLongPressUp() {
+    _currentSplash?.confirm();
+    _currentSplash = null;
+    widget.onLongPressUp?.call();
```
This seems like a behavioral change, and a good one. Can you find a way to test it, so that we can make sure it's not broken in the future?

houssemeddinefadhli81 replied:
@dkwingsmt I think the tree status has failed on the master branch

houssemeddinefadhli81 replied:
As always, @dkwingsmt , thanks for the great feedback! I’ll work on it..

Commit:
Update packages/flutter/lib/src/material/ink_well.dart

Co-authored-by: Tong Mu <dkwingsmt@users.noreply.github.com>

Commit:
Merge branch 'master' into add-onLongPressUp-to-inkwell

dkwingsmt reviewed (approved):
LGTM
